### PR TITLE
Several input formats supported by 7zip were added

### PR DESCRIPTION
**Pull request by arttarasov**
Unfortunately, it was tested only with SquashFS file format.
The list of supported formats was found [here](https://github.com/FauxFaux/7-zip/blob/master/CPP/7zip/Guid.txt)

### CODE REVIEW

**Comment by tomap** [comment]
Anyone else could confirm some of those new formats ?

**Comment by arttarasov** [comment]
Here is the cite from the official [7z source code](http://www.7-zip.org/download.html) (file CPP/7zip/Guid.txt):

```
Handler GUIDs:

{23170F69-40C1-278A-1000-000110xx0000}

  01 Zip
  02 BZip2
  03 Rar
  04 Arj
  05 Z
  06 Lzh
  07 7z
  08 Cab
  09 Nsis
  0A lzma
  0B lzma86
  0C xz
  0D ppmd

  C7 Ext
  C8 VMDK
  C9 VDI
  CA Qcow
  CB GPT
  CC Rar5
  CD IHex
  CE Hxs
  CF TE
  D0 UEFIc
  D1 UEFIs
  D2 SquashFS
  D3 CramFS
  D4 APM
  D5 Mslz
  D6 Flv
  D7 Swf
  D8 Swfc
  D9 Ntfs
  DA Fat
  DB Mbr
  DC Vhd
  DD Pe
  DE Elf
  DF Mach-O
  E0 Udf
  E1 Xar
  E2 Mub
  E3 Hfs
  E4 Dmg
  E5 Compound
  E6 Wim
  E7 Iso
  E8 
  E9 Chm
  EA Split
  EB Rpm
  EC Deb
  ED Cpio
  EE Tar
  EF GZip
```

Actually, these formats aren't new. For example, SquashFS support that I've tested was added to 7-Zip 9.18 beta at 2010-11-02 according to [the official release notes](http://www.7-zip.org/history.txt)

**Comment by squid-box** [comment]
If this is something you'd still want added, create a PR at https://github.com/squid-box/SevenZipSharp and I'll try to add it - like tomap I'd prefer more testing, but there are already several formats I've not been able to test properly, so why not add some more? :)

File: SevenZip/ArchiveEmulationStreamProxy.cs [hunk 1]
```diff
@@ -30,41 +30,23 @@ public ArchiveEmulationStreamProxy(Stream stream, int offset)
...
-            get
-            {
-                return Source.Position - Offset;
-            }
-            set
-            {
-                Source.Position = value;
-            }
+            get => Source.Position - Offset;
+            set => Source.Position = value;
```
Comment (mbobka) [inline SevenZip/ArchiveEmulationStreamProxy.cs:49]: there should be `value + Offset` or getter is wrong

**Comment by mbobka** [comment]
PR name is completely wrong. This PR mostly about unit testing.

**Comment by mbobka** [comment]
... and files formatting. Very important things were made by this PR, like:
```
       public int ChunkSize
       {
            get
            {
                return (int) _buffer.Length;
            }
        }
```
was replaced by:
```
        public int ChunkSize => (int) _buffer.Length;
```
This is very relevant to PR purpose!

**Comment by mbobka** [comment]
I recommend to reject the PR.

**Comment by squid-box** [comment]
@mbobka: somehow this pull request has the same activity as the one arttarasov opened on my fork. As is clearly seen his commit is the one related to the PR title, the rest is the entirety of my fork history.

Tomap has ceased development of this fork anyway, so I wouldn't worry to much.

**Pull request closed without merging**